### PR TITLE
Overhaul GitHub workflow to be reused across repos

### PR DESCRIPTION
**Pull request by nsoranzo**
Merge the 4 very similar workflows into a reusable one.

Also:
- Actually define `python-version` (3.11, as env) and install it
- Split out some steps from the "Run multi-line script"
- Install stable Planemo from PyPI instead of GitHub master branch
- Use `GITHUB_TOKEN` environment variable instead of `gh auth login`
- Reduce number of the slow `gh pr list` calls from 4 to 2 per repo
- Quote environment variables
- Update action versions
- Remove use of the ssh-agent action
- Allow to run the workflow manually
- Remove unused file

### CODE REVIEW

File: .github/workflows/autoupdate.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+              if [ "$(git branch -a --list "origin/$REPO")" != "" ]; then
+                echo "Branch exists without an open PR - deleting"
+                git push origin --delete "$REPO"
+              fi
+              echo "Creating branch and checking out"
+              git checkout -b "$REPO" "upstream/${{ matrix.upstream_repo_branch }}"
+            fi
+            echo "Running autoupdate command..."
+            cd "$REPO"
+            # --test --update_test_data
```
Comment (bernt-matthias) [inline .github/workflows/autoupdate.yml:98]: Can this be removed

```suggestion
```

**Review by bernt-matthias — APPROVED** [review]
Looks good to me. Will this also restore `planemo-autoupdate` as PR author (at the moment its `lldelisle`)?

**Comment by nsoranzo** [comment]
> Looks good to me. Will this also restore `planemo-autoupdate` as PR author (at the moment its `lldelisle`)?

I believe this is due to the PAT secret being used, which comes from Lucille.
Who has login access to the planemo-autoupdate GitHub account? @simonbray ?

**Comment by bgruening** [comment]
I converted it into an organization... you should have access. 

grafik

**Pull request merged**